### PR TITLE
create new writeread and cr examples

### PR DESCRIPTION
**Pull request by MichaelBrim**
### Description

This is the first step in cleaning up our examples by
improving code reuse. New headers have been created
that extend the functionality of the previous testlib.h.

The new writeread example improves upon sysio-writeread
and uses command-line switches to modify its behavior,
allowing the user to select the style of write/read.
Supported styles include standard write|read, pwrite|pread,
async I/O, and lio_listio. The example also supports modes
of operation where data is checked on read, or different
data is read than written.

The new cr example, short for checkpoint-restart, does
multiple checkpoints before laminating the target file,
then re-opens the file before reading the restart data.
The cr example also supports the varied write/read
styles.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the UnifyCR code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted.

### CODE REVIEW

**Comment by MichaelBrim** [comment]
I'm working on cleaning up the checkpatch and 'make dist' errors

File: examples/src/testutil_rdwr.h [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+    off_t off;
+    void* dst;
+
+    assert(NULL != cfg);
+
+    errno = 0;
+    if (cfg->use_aio) { // aio_read(2)
+        rc = aio_read(req);
+        if (-1 == rc) {
+            test_print(cfg, "aio_write() failed");
```
[the remark this answers is not in the document]
Comment (MichaelBrim) [reply examples/src/testutil_rdwr.h]: I'll fix this.

**Comment by MichaelBrim** [comment]
@CamStan Yes, the eventual goal is to remove testlib.h and covert the other sysio examples to use testutil and the support for multiple read/write methods.

**Comment by MichaelBrim** [comment]
@CamStan can you review this again. I accidentally pushed some other commits related to implementing missing POSIX I/O support that were originally destined for a separate PR (now closed).

**Comment by CamStan** [comment]
Not a problem. I'll give it another pass.

**Pull request merged**